src/Runtime/Utils.fs

```diff
@@ -36,32 +36,56 @@ type Utils private() =
     
     [<Extension>]
     [<EditorBrowsable(EditorBrowsableState.Never)>]
-    static member MapRowValues<'TItem>(cursor: DbDataReader, resultType : ResultType, resultSet : ResultSetDefinition) =
+    static member MapRowValues<'TItem>(cursor: DbDataReader, resultType : ResultType, resultSet : ResultSetDefinition, isTypeReuseEnabled) =
         let rowMapping =
             if resultSet.ExpectedColumns.Length = 1 then
                 Array.item 0
             elif resultType = ResultType.Tuples then
                 let clrTypeName = resultSet.SeqItemTypeName
                 Reflection.FSharpValue.PreComputeTupleConstructor(Type.GetType(clrTypeName, throwOnError = true))
             else
-                box
+                Array.copy >> box
```

Comment:
@melanore, this is an absolutely critical fix. Because the array is reused for reads of multiple rows and not copied here, every record in a list of results is the same object.